### PR TITLE
Cortex-M code changed to support IAR compiler

### PR DESCRIPTION
**Pull request by SparkiDev**

### CODE REVIEW

File: configure.ac [hunk 1]
```diff
@@ -4039,8 +4039,8 @@ if test "$ENABLED_SP_ASM" = "yes"; then
       ENABLED_SP_ARM_THUMB_ASM=yes
     else
       if test $host_alias = "cortex"; then
-        AM_CFLAGS="$AM_CFLAGS -DWOLFSSL_SP_ARM_CORTEX_ASM"
-        AM_CCASFLAGS="$AM_CCASFLAGS -DWOLFSSL_SP_ARM_CORTEX_ASM"
+        AM_CFLAGS="$AM_CFLAGS -DWOLFSSL_SP_ARM_CORTEX_M_ASM -mcpu=cortex-m4"
```
Comment (dgarske) [inline configure.ac]: Does it make sense to assume `cortex-m4` here? What about the `cortex-m3` or other M variants?

File: wolfcrypt/src/sp_cortexm.c [hunk 2]
```diff
@@ -190,500 +190,500 @@ SP_NOINLINE static void sp_2048_mul_8(sp_digit* r, const sp_digit* a,
         const sp_digit* b)
 {
     sp_digit tmp[8];
-    __asm__ __volatile__ (
+    asm volatile (
```
Comment (dgarske) [inline wolfcrypt/src/sp_cortexm.c]: Is this change for `asm` and `volatile` okay for all compilers? Perhaps we should have macros for these and detect the `__IAR_SYSTEMS_ICC__` case?

File: configure.ac (hunk 1 above)
Comment (SparkiDev) [reply configure.ac]: gcc won't compile assembly code unless something is chosen.
cortex-m4 is what I'm testing and users will most likely have there own user_settings.h.

File: wolfcrypt/src/sp_cortexm.c (hunk 2 above)
Comment (SparkiDev) [reply wolfcrypt/src/sp_cortexm.c]: Changed as requested.

**Pull request merged**